### ISSUE TITLE
Комментарии к первоначальной заявке

### ISSUE THREAD

**Issue by aakuptsov** [kotfind/learning_graph#1]
1. Для лучшего обоснования проблемного поля необходимо привести несколько конкретных кейсов обучения, в которых проявляется рассматриваемая проблема.

2. Оказывает ли специфика изучаемой темы на возможности применения графов обучения? (например, изучать можно готский язык, вязание крючком, дифференциальные уравнения и т. п.)

3. Хотя, возможно, полных аналогов программных продуктов и нет, нужно проанализировать смежные продукты, например, веб-сервисы, позволяющие рисовать графы, и указать, почему с их помощью нельзя решить проблемное поле. 

4. Насколько я понимаю, это будет десктоп-приложение? Чем обусловлена подобная форма существования продукта?

5. Как будет осуществляться работа с Википедией и другими сервисами? С помощью чего?

**Comment by kotfind** [comment]
4. Да, это будет десктоп-приложение. Выбор не пал на web-сервис, так как хочется создать приложение, которое не зависело бы от наличия постоянного доступа в интернет у клиента или работоспособности серверов у поставщика. Выбирая же между приожением для компьютера или для смартфона, первое показалось мне предпочтительным, так как предполагается, что во время учебы пользователь не будет перемещаться (например, будет сидеть у себя дома, в учебном заведении или библиотеке) и в такой ситуации, как мне кажется, использовать компьютер будет более удобно.

Ответы на остальные вопросы включил непосредственно в файл с заявкой.

**Comment by aakuptsov** [comment]
1. А какой в продукте будет бэкенд? Проекты без БД к защите не допускаются.

**Comment by kotfind** [comment]
Бэкенд предполагается написать на C++. В качестве БД будет использоваться MySQL.

**Comment by aakuptsov** [comment]
ЗАЧЁТ

**Comment by kotfind** [comment]
Хотелось бы внести некоторые изменения в заявку (<s>а после, соответственно, и в пользовательские сценарии</s> upd: уже внесены). Измененный файл `..._aplication.md` загружен в репозиторий в ветку `toConfirm`. Основные направления изменений:

1. **Исправление неточностей**, а именно: опечаток и неточностей, появившихся из-за меньшего владения необходимыми знаниями на момент написания заявки, чем сейчас.
1. **Добавление учителей, репетиторов и тд в целевую аудиторию.** Добаление соответствующих изменений в "проблемное поле".
1. **Добавлене возможности изменять размер шрифта и язык приложения.**
1. **Превращение графа в полноценную сущность:** в прошлой версии заявки граф являлся временной сущностью только для вывода информации. Теперь же графы можно создавать, хранить, удобно редактировать и т.д. (Без этого программа казалось мне неполной.)
1. **Уничтожение списков "изучить позже" и "изученные", как отдельных сущностей.** Теперь списки не имеют собственной вкладки. Для их просмотра нужно выбрать соответствующие фильтры при поиске тем. (Предполагаю, что так взаимодействовать с ними будет удобнее и интуитивее.)
1. **Уничтожение функции автоматической генерации.** (Предполагаю, что сделать генерацию на том уровне, на котором в ней будет смысл, у меня не получится. Надеюсь, что, учитывая некоторые из указанных выше изменений, даже без этой функции проект не станет считаться чересчур простым.)

Соответственно, хотелось бы получить подверждение на внесение этих изменений.

**Comment by aakuptsov** [comment]
К сожалению, мы не можем одобрить отсутствие пункта №6, поскольку это был один из тех пунктов, которые сделали проект "защищабельным" и интересным. В качестве компромисса предлагаем, например, генерировать граф не по всем упомянутым сайтам, а по одному. 

1 - OK
2 - OK
3 - OK
4 - OK, очень хорошо
5 - я так понял, это не отмена функционала, а изменение его реализации? Тогда OK

**Comment by kotfind** [comment]
> 5 - я так понял, это не отмена функционала, а изменение его реализации?

Да, Вы правильно поняли. Раньше пользователь открывал отдельную вкладку "Списки", а теперь он будет выбирать соответствующую галочку во вкладке "Темы".

> К сожалению, мы не можем одобрить отсутствие пункта №6, поскольку это был один из тех пунктов, которые сделали проект "защищабельным" и интересным. В качестве компромисса предлагаем, например, генерировать граф не по всем упомянутым сайтам, а по одному.

Про 6-ое понял &mdash; подумаю.

**Comment by kotfind** [comment]
Добрый день. Подумал над шестым пунктом и пришел к следующим соображениям. Сделать полностью автоматическую генерацию графа мне всё ещё не кажется возможным: основная проблема заключается в том, что существующие источники не предоставляют всей необходимой для этого информации, и получать её не от куда. С другой стороны, мне пришло в голову несколько похожих идей, которые, как мне кажется, могли бы подойти.

**Первая идея:** Полуавтоматическая генерация графа.

Как я говорил выше, существующие источники не содержат всей необходимой для построения графа информации &mdash; этот вариант предполагает взаимодействие с пользователем для её получения.

Таким образом, для генерации графа пользователь должен будет ввести ссылку на статью с википедии, тогда система превратит эту статью в тему. После система просмотрит все гиперссылки на этой странице википедии и создаст из них темы, соединив их связью с изначальной темой. И так далее рекурсивно.

Во время генерации система будет спрашивать пользователя о направлении связи (т.е. о том, какая тема от какой зависит), ибо узнать это автоматически не представляется возможным.

**Вторая идея:** Автоматическая загрузка описания темы.

В данном варианте при создании темы система будет предлагать пользователю автоматически загрузить её описание.

Говоря более конкретно, после того, как пользователь введет название темы, система проверит википедию и ещё несколько источников на предмет наличия информации о данной теме. Все обнаруженные варианты будут предложены пользователю. Пользователь сможет выбрать один или несколько вариантов, которые будут автоматически добавлены системой в качестве описания данной темы.

**Итого** хотелось бы узнать, будет ли какой-либо из этих вариантов достаточным для замены 6-ого пункта?

**Comment by aakuptsov** [comment]
**Первая идея: Полуавтоматическая генерация графа.** - ОК, одобряем первую идею.
